code/modules/antagonists/revenant/revenant.dm

```diff
@@ -345,6 +347,40 @@
 	alpha=255
 	stasis = FALSE
 
+/mob/living/simple_animal/revenant/orbit(atom/target)
+	setDir(SOUTH)//reset dir so the right directional sprites show up
+	return ..()
+
+/mob/living/simple_animal/revenant/Moved(atom/OldLoc)
+	if(!orbiting) // only needed when orbiting
+		return ..()
+	if(incorporeal_move_check(src))
+		return ..()
+
+	orbiting?.end_orbit(src)
+	abstract_move(OldLoc) // back back back it up, gross but maybe someday there'll be a jaunt orbit component
+
+/mob/living/simple_animal/revenant/stop_orbit(datum/component/orbiter/orbits)
+	animate(src, pixel_y = 2, time = 1 SECONDS, loop = -1, flags = ANIMATION_RELATIVE)
+	animate(pixel_y = -2, time = 1 SECONDS, flags = ANIMATION_RELATIVE)
+	return ..()
+
+/// Incorporeal move check: blocked by holy-watered tiles and salt piles.
+/mob/living/simple_animal/revenant/proc/incorporeal_move_check(atom/destination)
+	var/turf/open/floor/stepTurf = get_turf(destination)
+	if(stepTurf)
+		for(var/obj/effect/decal/cleanable/food/salt/S in stepTurf)
```

Comment:
```suggestion
		if(locate(/obj/effect/decal/cleanable/food/salt) in stepTurf)
```

Reply:
Should I update the hotter move proc code where I stole this from too?
https://github.com/tgstation/tgstation/blob/1f08928e211d904c40f228529b7186548d6571a2/code/modules/mob/mob_movement.dm#L251

Reply:
No harm in doing it!

Reply:
okay sure, also please add Code Improvement label